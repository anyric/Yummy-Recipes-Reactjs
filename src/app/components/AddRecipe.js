import React, { Component } from 'react';

import * as Utility from '../utils/Utility';

const url = 'http://127.0.0.1:5000/recipe/api/v1.0/allcategory/';

export class AddRecipe extends Component {
  constructor(props) {
    super(props);
    Utility.getCategory(url);
    this.state = {
      categorylist: JSON.parse(localStorage.getItem('allcategory')),
    };
  }

  handleInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    this.setState({
      [name]: value,
    });
  }
  render() {
    return (
      <div className="container-fluid dborder mt-5 col-sm-8 offset-sm-2 col-md-8 offset-md-1 pt-3">
        <h2>Add Recipe</h2>
        <hr />
        <form onSubmit={Utility.addRecipe.bind(this)}>
          <div className="container">
            <div className="col-xs-7">
              <p>Please fill in this form to add a new Recipe.</p>
            </div>
            <div className="form-group col-xs-7">
              <label htmlFor="sel1">Select Category:</label>
              <select
                className="form-control"
                name="catid"
                onChange={this.handleInputChange.bind(this)}
              >
                {this.state.categorylist ? this.state.categorylist.map((category, index) =>
                  <option value={category.id} key={index}>{category.name}</option>) :
                <option>No Category</option>
                }
              </select>
            </div>
            <div className="form-group col-xs-7">
              <label htmlFor="name">Name:</label>
              <input
                type="text"
                className="form-control"
                id="fullname"
                name="recname"
                placeholder="Enter Name"
                onChange={this.handleInputChange.bind(this)}
              />
            </div>
            <div className="form-group col-xs-7">
              <label htmlFor="ingredient">Ingredients:</label>
              <textarea
                className="form-control"
                row="5"
                id="ingredient"
                name="recdesc"
                placeholder="Enter Ingredients"
                onChange={this.handleInputChange.bind(this)}
              >
              </textarea>
            </div>
            <div className="col-xs-10">
              <button type="submit" className="btn btn-primary btn-md">
                <span className="glyphicon glyphicon-floppy-save">    Save          </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    );
  }
}
