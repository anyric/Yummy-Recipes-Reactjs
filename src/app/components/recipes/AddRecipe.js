import React, { Component } from 'react';

import axios from 'axios';

import * as Recipes from '../../controller/Recipes';

const url = 'http://127.0.0.1:5000/recipe/api/v1.0/allcategory/';
axios.defaults.headers = { 'Content-Type': 'application/json' };

export class AddRecipe extends Component {
  constructor() {
    super();
    this.getCategory();
    this.state = { categorylist: [] };
  }
  getCategory() {
    const config = { headers: { 'x-access-token': localStorage.getItem('token') } };
    const self = this;
    axios.get(url, config)
      .then(function (res) {
        self.setState({ categorylist: res.data });
      })
      .catch(function (error) {
        if (error.response) {
          alert(error.response.data.message);
        }
      });
  }
  handleInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    this.setState({
      [name]: value,
    });
  }
  render() {
    return (
      <div className="container-fluid dborder mt-5 col-sm-8 offset-sm-2 col-md-8 offset-md-1 pt-3">
        <h2>Add Recipe</h2>
        <hr />
        <form onSubmit={Recipes.addRecipe.bind(this)}>
          <div className="container">
            <div className="col-xs-7">
              <p>Please fill in this form to add a new Recipe.</p>
            </div>
            <div className="form-group col-xs-7">
              <label htmlFor="sel1">Select Category:</label>
              <select
                className="form-control"
                name="catid"
                onChange={this.handleInputChange.bind(this)}
              >
                {this.state.categorylist ? this.state.categorylist.map((category, index) =>
                  <option value={category.id} key={index}>{category.name}</option>) :
                <option>No Category</option>
                }
              </select>
            </div>
            <div className="form-group col-xs-7">
              <label htmlFor="name">Name:</label>
              <input
                type="text"
                className="form-control"
                id="fullname"
                name="recname"
                placeholder="Enter Name"
                onChange={this.handleInputChange.bind(this)}
              />
            </div>
            <div className="form-group col-xs-7">
              <label htmlFor="ingredient">Ingredients:</label>
              <textarea
                className="form-control"
                row="5"
                id="ingredient"
                name="recdesc"
                placeholder="Enter Ingredients"
                onChange={this.handleInputChange.bind(this)}
              >
              </textarea>
            </div>
            <div className="col-xs-10">
              <button type="submit" className="btn btn-primary btn-md">
                <span className="glyphicon glyphicon-floppy-save">    Save          </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    );
  }
}
