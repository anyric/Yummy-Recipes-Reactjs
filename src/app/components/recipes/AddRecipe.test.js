import React from 'react';
import { shallow } from 'enzyme';

import { AddRecipe } from './AddRecipe';

describe('AddRecipe component', () => {
  const component = shallow(<AddRecipe />);
  it('renders without crashing', () => {
    shallow(<AddRecipe />);
  });
  it('returns one component', () => {
    expect(component).toHaveLength(1);
  });
  it('renders properly', () => {
    expect(component).toMatchSnapshot();
  });
  it('renders divs', () => {
    expect(component.find('div').length).toBe(7);
  });
});

