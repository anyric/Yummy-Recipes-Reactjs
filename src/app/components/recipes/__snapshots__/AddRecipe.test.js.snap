// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRecipe component renders properly 1`] = `
<div
  className="container-fluid dborder mt-5 col-sm-8 offset-sm-2 col-md-8 offset-md-1 pt-3"
>
  <h2>
    Add Recipe
  </h2>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <div
      className="container"
    >
      <div
        className="col-xs-7"
      >
        <p>
          Please fill in this form to add a new Recipe.
        </p>
      </div>
      <div
        className="form-group col-xs-7"
      >
        <label
          htmlFor="sel1"
        >
          Select Category:
        </label>
        <select
          className="form-control"
          name="catid"
          onChange={[Function]}
        >
          <option>
            No Category
          </option>
        </select>
      </div>
      <div
        className="form-group col-xs-7"
      >
        <label
          htmlFor="name"
        >
          Name:
        </label>
        <input
          className="form-control"
          id="fullname"
          name="recname"
          onChange={[Function]}
          placeholder="Enter Name"
          type="text"
        />
      </div>
      <div
        className="form-group col-xs-7"
      >
        <label
          htmlFor="ingredient"
        >
          Ingredients:
        </label>
        <textarea
          className="form-control"
          id="ingredient"
          name="recdesc"
          onChange={[Function]}
          placeholder="Enter Ingredients"
          row="5"
        />
      </div>
      <div
        className="col-xs-10"
      >
        <button
          className="btn btn-primary btn-md"
          type="submit"
        >
          <span
            className="glyphicon glyphicon-floppy-save"
          >
                Save          
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
